crates/router/src/connector/bamboraapac/transformers.rs

```diff
@@ -1,234 +1,804 @@
 };
 
-//TODO: Fill the struct with respective fields
-pub struct BamboraapacRouterData<T> {
-    pub amount: MinorUnit, // The type of amount that a connector accepts, for example, String, i64, f64, etc.
-    pub router_data: T,
-}
+type Error = error_stack::Report<errors::ConnectorError>;
 
-impl<T> TryFrom<(MinorUnit, T)> for BamboraapacRouterData<T> {
-    type Error = error_stack::Report<errors::ConnectorError>;
-    fn try_from((amount, item): (MinorUnit, T)) -> Result<Self, Self::Error> {
-        //Todo :  use utils to convert the amount to the type of amount that a connector accepts
-        Ok(Self {
-            amount,
-            router_data: item,
-        })
-    }
-}
+// request body in soap format
+pub fn get_payment_body(req: &types::PaymentsAuthorizeRouterData) -> Result<Vec<u8>, Error> {
+    let transaction_data = get_transaction_body(req)?;
+    let body = format!(
+        r#"
+            <soapenv:Envelope xmlns:soapenv="http://schemas.xmlsoap.org/soap/envelope/" 
+            xmlns:dts="http://www.ippayments.com.au/interface/api/dts">
+                <soapenv:Body>
+                    <dts:SubmitSinglePayment>
+                        <dts:trnXML>
+                            <![CDATA[
+                                {}
+                            ]]>    
+                        </dts:trnXML>
+                    </dts:SubmitSinglePayment>
+                </soapenv:Body>
+            </soapenv:Envelope>
+        "#,
+        transaction_data
+    );
 
-//TODO: Fill the struct with respective fields
-#[derive(Default, Debug, Serialize, Eq, PartialEq)]
-pub struct BamboraapacPaymentsRequest {
-    amount: MinorUnit,
-    card: BamboraapacCard,
+    Ok(body.as_bytes().to_vec())
 }
 
-#[derive(Default, Debug, Serialize, Eq, PartialEq)]
-pub struct BamboraapacCard {
-    number: cards::CardNumber,
-    expiry_month: Secret<String>,
-    expiry_year: Secret<String>,
-    cvc: Secret<String>,
-    complete: bool,
+fn get_transaction_body(req: &types::PaymentsAuthorizeRouterData) -> Result<String, Error> {
+    let amount = req.request.get_amount();
+    let auth_details = BamboraapacAuthType::try_from(&req.connector_auth_type)?;
+    let transaction_type = get_transaction_type(req.request.capture_method)?;
+    let card_info = get_card_data(req)?;
+    let transaction_data = format!(
+        r#"
+        <Transaction>
+            <CustRef>{}</CustRef>
+            <Amount>{}</Amount>
+            <TrnType>{}</TrnType>
+            <AccountNumber>{}</AccountNumber>
+            {}
+            <Security>
+                    <UserName>{}</UserName>
+                    <Password>{}</Password>
+            </Security>
+        </Transaction>
+    "#,
+        req.connector_request_reference_id.to_owned(),
+        amount,
+        transaction_type,
+        auth_details.account_number.peek(),
+        card_info,
+        auth_details.username.peek(),
+        auth_details.password.peek(),
```

Comment:
We will log these username and password in `raw_connector_request`, can we mitigate this somehow? e.g making the complete transaction_data string as secret